src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -732,9 +732,10 @@ private static void WriteRandomMask(byte[] buffer, int offset)
             }
             catch (Exception exc)
             {
-                throw _state == WebSocketState.Aborted ?
-                    new WebSocketException(WebSocketError.InvalidState, SR.Format(SR.net_WebSockets_InvalidState_ClosedOrAborted, "System.Net.WebSockets.InternalClientWebSocket", "Aborted"), exc) :
-                    new WebSocketException(WebSocketError.ConnectionClosedPrematurely, exc);
+                if (_state == WebSocketState.Aborted) {
+                    throw new OperationCanceledException("Aborted");
```

Comment:
- Please pass the `exc` as the InnerException argument to the OCE's ctor, so that the relevant failure isn't lost.
- Please change the `"Aborted"` to instead be `nameof(WebSocketState.Aborted)`.
- Nit: Move the brace to be at the start of its own line